style_guide.md

```diff
@@ -0,0 +1,118 @@
+
+### Inline Comments
+You should document all loop and control structures with inline comments. 
+These comments should be indented at least 8 spaces after the end of the line. 
+Indentation of the comment should match indentation of the line it describes. 
+There should be a space after each '#', and the first letter of the comment should be captialized. 
+Ex. 
+```python
+while self.isAlive is True:                # While person is alive
+    if self.age < 6 or self.isRetired:         # If too old or young
+        stay_home()                                # Stay home
+    elif self.age < 18:                        # Else if school age
+        goto_school()                              # Goto school
+    else:                                      # Otherwise
+        goto_work()                                # Goto work
+```
+
+### Block Comments 
+Block comments should be used to describe logical groupings of statements, if appropriate, but not loop or control structures. 
+They should be placed above the code block they describe, with a new line above itself and below the code block it describes. 
+Ex. 
+```python
+# Do the work
+work_it_harder()
+make_it_better()
+do_it_faster()
+makes_us_stronger()
+
+# Ramble on afterwards
+more_than_ever()
+hour_after()
+our_work_is()
+never_over()
+
+end_song()
+```
+## 4. Docstrings
+Docstrings should follow the [Google Python](http://google.github.io/styleguide/pyguide.html?showone=Comments#Comments) format.
+
+### Single Line Summaries
+If you are having trouble writing a single line summary in a single line; the reason usually boils down to **"This X is too complicated."** 
+
+Try these solutions...
+1. **Describe *what* X represents or does, not what X *can* do or *how*.** 
+2. **Refactor X into X and Y.** A good rule in programming is to have each *thing* do one, and only one thing. Classes should represent one object, and functions should accomplish one task. 
+
+If you can't find a good way to do either; consider opening an issue about it, or commenting on it with your PR.
+
+### Docstring Format
+Docstrings **must** immediately follow a class or function definition. 
+There should be a new line immediately after the single line summary, and immediately before the first section (if there is additional info).
+
+```
+DOCSTRING FORMAT
+KEY = [STUFF] = required data, <STUFF> = Data required if available, |STUFF| = optiondal data, [/] = new line
+```
+```python
+def MyClass(object):
+    """[Single line summary of class.]
+    [/]
+    |Additional info if relevant and useful.|
+    |/|
+    <Args:
+        arg1: Description of parameter.
+        ...>
+    [/]
+    <Attributes:
+        attr1: Description of attribute. 
+        ...>
+    [/]
+    <Returns:
+        return_value: Data type and description of return_value.
+        ...>
+    [/]
+    <Yields:
+        yield_value: Description of yield_value. 
+        ...>    
+    """
+```
```

Comment:
I think we can maybe move these two sections around, so that we show the format first, and then give the information about the one line summary.